src/puppetlabs/comidi.clj

```diff
@@ -167,11 +167,13 @@
     (update-in route-info [:path] conj "!")
 
     (sequential? pattern)
-    (if-let [next (first pattern)]
-      (update-route-info*
-        (update-in route-info [:path] conj next)
-        (rest pattern))
-      route-info)
+    (loop [info route-info
+           [head & tail] pattern]
+        (if (nil? tail)
```

Comment:
Generally when an explicit nil check is used, the conditional should be reversed and changed to `(if thing ...)`

```clojure
(if tail
(recur...)
(update-in...))
```

Reply:
Works for me, now that this got switched from `empty?` to `nil?`.